Pass structure to GenerateConditions functor to avoid duplication.

justinmichaud opened the pull request:
#### 19efd3c1612a963a9d1a7d0dfcc03dbace57a540
<pre>
Pass structure to GenerateConditions functor to avoid duplication.
<a href="https://bugs.webkit.org/show_bug.cgi?id=241930">https://bugs.webkit.org/show_bug.cgi?id=241930</a>

Reviewed by Saam Barati.

* Source/JavaScriptCore/bytecode/ObjectPropertyConditionSet.cpp:
(JSC::generateConditionsForPropertyMiss):
(JSC::generateConditionsForPropertySetterMiss):
(JSC::generateConditionsForPrototypePropertyHit):
(JSC::generateConditionsForPrototypePropertyHitCustom):
(JSC::generateConditionsForInstanceOf):
(JSC::generateConditionsForPrototypeEquivalenceConcurrently):
(JSC::generateConditionsForPropertyMissConcurrently):
(JSC::generateConditionsForPropertySetterMissConcurrently):
(JSC::generateConditionForSelfEquivalence):

Canonical link: <a href="https://commits.webkit.org/251871@main">https://commits.webkit.org/251871@main</a>
</pre>

saambarati commented on Source/JavaScriptCore/bytecode/ObjectPropertyConditionSet.cpp:
```diff
@@ -179,9 +179,11 @@ static constexpr bool verbose = false;
 {
-    Structure* structure = object->structure();
+    if (object->structure() != structure)
+        return ObjectPropertyCondition();
```
Let's not add these branches. They're a bit misleading about this structure not changing. I'd vote for a comment or encoding Structure Object pair in a type.

saambarati commented on Source/JavaScriptCore/bytecode/ObjectPropertyConditionSet.cpp:
```diff
@@ -289,11 +291,14 @@ ObjectPropertyConditionSet generateConditions(JSGlobalObject* globalObject, Stru
         }
+
+        if (structure != object->structure())
+            return ObjectPropertyConditionSet::invalid();
```
Let's not add these branches. They're a bit misleading about this structure not changing. I'd vote for a comment or encoding Structure Object pair in a type.

saambarati reviewed (approved):
r=me

Commit:
Pass structure to GenerateConditions functor to avoid duplication.
https://bugs.webkit.org/show_bug.cgi?id=241930

Reviewed by Saam Barati.

* Source/JavaScriptCore/bytecode/ObjectPropertyConditionSet.cpp:
(JSC::generateConditionsForPropertyMiss):
(JSC::generateConditionsForPropertySetterMiss):
(JSC::generateConditionsForPrototypePropertyHit):
(JSC::generateConditionsForPrototypePropertyHitCustom):
(JSC::generateConditionsForInstanceOf):
(JSC::generateConditionsForPrototypeEquivalenceConcurrently):
(JSC::generateConditionsForPropertyMissConcurrently):
(JSC::generateConditionsForPropertySetterMissConcurrently):
(JSC::generateConditionForSelfEquivalence):

Canonical link: https://commits.webkit.org/251871@main
